Spanner/src/Connection/Grpc.php

```diff
@@ -1194,6 +1156,40 @@ public function commit(array $args)
         ]);
     }
 
+    /**
+     * @param array $args
+     * @return \Generator
+     */
+    public function batchWrite(array $args)
+    {
+        $databaseName = $this->pluck('database', $args);
+        $mutationGroups = $this->pluck('mutationGroups', $args);
+        $requestOptions = $this->pluck('requestOptions', $args, false) ?: [];
+
+        array_walk(
+            $mutationGroups,
+            fn(&$x) => $x['mutations'] = $this->parseMutations($x['mutations'])
+        );
+
+        array_walk($mutationGroups, fn(&$x) => $x = $this->serializer->decodeMessage(
+            new BatchWriteRequestMutationGroup,
+            $x
+        ));
```

Comment:
This should use `array_map` instead of `array_walk` with pass-by-reference
```suggestion
        $mutationGroups = array_map(fn($x) => $this->serializer->decodeMessage(
            new BatchWriteRequestMutationGroup,
            $x
        ), $mutationGroups);
```